lua/hawtkeys/ts.lua

```diff
@@ -341,24 +341,43 @@ local function find_maps_in_file(filePath)
                     )
                     break
                 end
+                local unknown_value = { "unknown" }
                 local strObj =
                     vim.treesitter.get_node_text(node.node, fileContent)
                 local ok, tableObj = pcall(function()
+                    local env = setmetatable({}, {
+                        __index = function()
+                            setmetatable(unknown_value, {
+                                __index = function()
+                                    return unknown_value
+                                end,
+                            })
+                            return unknown_value
+                        end,
+                    })
                     --Remove wrapping parens and wrap in table and unpack - issue #81
                     strObj = strObj:gsub("^%s*%(%s*", ""):gsub("%s*%)%s*$", "")
-                    return loadstring("return {" .. strObj .. "}")()
+                    local func = loadstring("return {" .. strObj .. "}")
+                    if not func then
+                        return {}
+                    end
+                    setfenv(func, env)
+                    local obj = func()
+                    return obj
                 end)
                 if not ok then
                     vim.notify_once(
                         "Error parsing which-key table",
                         vim.log.levels.ERROR
                     )
                     vim.notify(strObj, vim.log.levels.ERROR)
-                    break
                 end
                 local wkMapping = which_key.parse(unpack(tableObj))
 
                 for _, mapping in ipairs(wkMapping) do
+                    if mapping.cmd == nil then
+                        mapping.cmd = "Function uses out of scope variables"
```

Comment:
I think this should go on a separate `error` prop that can be displayed if the main prop is nil. Feels a bit weird to just include an error message without any kind of flag for it.